src/aap_eda/api/views/auth.py

```diff
@@ -92,3 +96,27 @@ class SessionLogoutView(APIView):
     def post(self, request: Request, *args, **kwargs):
         logout(request)
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class TokenRefreshView(APIView):
+    permission_classes = ()
+    authentication_classes = ()
+
+    @extend_schema(
+        operation_id="token_refresh",
+        description="Refresh websocket access token",
+        request=RefreshTokenSerializer,
+        responses={
+            status.HTTP_200_OK: OpenApiResponse(JWTTokenSerializer),
+        },
+    )
+    @method_decorator(never_cache)
+    def post(self, request: Request, *args, **kwargs):
+        serializer = RefreshTokenSerializer(data=request.data)
```

Comment:
@bzwei without the serializer couldn't we just call 
```
   user = validate_jwt_token(data["refresh"], "refresh")
   token = jwt_access_token(user.id)
```
once we check if there is a refresh in the payload

Reply:
We still need the serializer to do validation. I moved the call to create new token to the view.